### PR TITLE
Add guide for Rocket.Chat

### LINKED ISSUE #584: Guide for Rocket.Chat
Hi, I'm currently writing a guide for [Rocket.Chat](https://rocket.chat/).

### PR DESCRIPTION
**Pull request by d-sko**
Hi, I've added a guide for [Rocket.Chat](https://rocket.chat/).

closes #584

### CODE REVIEW

File: source/guide_rocketchat.rst [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+   --bind_ip 127.0.0.1
+   --auth
+   --unixSocketPrefix %(ENV_HOME)s/mongodb
+   --replSet rs01
+ autostart=yes
+ autorestart=yes
+
+This is required because Rocket.Chat uses Meteor Oplog Tailing for performance improvements (see `the docs <https://rocket.chat/docs/installation/manual-installation/mongo-replicas/>`_ for further information). 
+
+Then tell supervisord to update and start the service:
```
Comment (luto) [inline source/guide_rocketchat.rst]: This probably also needs `supervisorctl reread` to reload the config file.

File: source/guide_rocketchat.rst [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+Setup daemon
+------------
+
+Create ``~/etc/services.d/rocket.chat.ini`` with the following content (don't forget to use your URL for ``ROOT_URL``):
+
+.. code-block:: ini
+
+ [program:rocket.chat]
+ command=node %(ENV_HOME)s/rocket.chat/main.js
+ environment=MONGO_URL="mongodb://<username>_rocketchat:<password>@localhost:27017/rocketchat?replicaSet=rs01&authSource=admin",MONGO_OPLOG_URL="mongodb://<username>_rocketchat:<password>@localhost:27017/local?replicaSet=rs01&authSource=admin",ROOT_URL="https://<username>.uber.space/",PORT=3000
```
Comment (luto) [inline source/guide_rocketchat.rst]: Please split this into multiple lines (one per var), like in the [synapse guide](https://lab.uberspace.de/guide_synapse.html#setup-daemon).

File: source/guide_rocketchat.rst [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+ [isabell@stardust ~]$ supervisorctl update
+ mongodb: stopped
+ mongodb: updated process group
+ [isabell@stardust ~]$ 
+
+Now initiate the replica set:
+
+::
+
+ [isabell@stardust ~]$ mongo --username <username>_mongoroot --eval "printjson(rs.initiate())"
```
Comment (luto) [inline source/guide_rocketchat.rst]: The mongodb guide suggest to use the uberspace username as `<username>`, so I think we can replace all `<username>_mongoroot` with `${USER}_mongoroot` to make the guide easier to follow.

File: source/guide_rocketchat.rst (hunk 2 above)
Comment (luto) [inline source/guide_rocketchat.rst]: How about using `%(ENV_USER)s_rocketchat` here to reduce the number of things to manually replace?

File: source/guide_rocketchat.rst [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+
+We're using :manual:`Node.js <lang-nodejs>` in the stable version 12:
+
+::
+
+ [isabell@stardust ~]$ uberspace tools version show node
+ Using 'Node.js' version: '12'
+ [isabell@stardust ~]$
+
+We'll also need :lab:`MongoDB <guide_mongodb>`, so follow the MongoDB guide and come back when it's running.
```
Comment (luto) [inline source/guide_rocketchat.rst]: How can we feature this more prominently? Maybe add it to the "familiar with the basic concepts of"-list? put it into a note box? List it as the first (or last) prerequisite?

File: source/guide_rocketchat.rst [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+
+Check your chat URL setup:
+
+::
+
+ [isabell@stardust ~]$ uberspace web domain list
```
Comment (luto) [inline source/guide_rocketchat.rst]: Just like web backends, there is also an include for this.

**Comment by d-sko** [comment]
Thanks for the feedback! I'll change the guide accordingly.

**Comment by d-sko** [comment]
Damn. I've fixed everything but upon testing I stumbled over https://github.com/RocketChat/Rocket.Chat/issues/7465, this didn't happen on my other tests. I'll update the PR when I find a reasonable solution.

**Comment by d-sko** [comment]
I think I've found a solution that's not too bad 😅 .

**Comment by luto** [comment]
Would you like to try to submit your `/tmp/ufs` change as a pull request to rocketchat? If at all possible, we do not want to force users to edit the source code. That increases the complexity and tends to break on updates because, well, the code changes :sweat_smile:

**Comment by d-sko** [comment]
Yes, fixing this in Rocket.Chat would be nicer of course. I just figured that a breaking change is unlikely here since the change is not in Rocket.Chat but the UploadFS package, which is deprecated (see https://github.com/jalik/meteor-jalik-ufs#uploadfs) and is unlikely to change much.

But regardless of that I'll have a look at the Rocket.Chat code... well, that escalated quickly :grin:

**Comment by off1ine** [comment]
Hi and thanks a lot for this guide - just what I was looking for :) 
Sorry for maybe breaking protocol, I'm still figuring out github. Is there a way you could add the content for the rocket.chat-user-setup.js file in line 193? I can see it in the raw code for the guide but it doesn't seem to show up for me (OS X FF, Chrome). Thanks a lot for your work!

**Comment by d-sko** [comment]
Hi, you can view the full guide in the netlify preview: https://deploy-preview-585--uberlab.netlify.com/guide_rocketchat.html.

**Comment by d-sko** [comment]
I've made a PR in Rocket.Chat, lets see what happens :sweat_smile: 

https://github.com/RocketChat/Rocket.Chat/pull/17012

**Comment by off1ine** [comment]
Perfect, thank you! It worked and I'm up and running :)

**Comment by luto** [comment]
Thank you very much for all the work and the upstream PR! :tada:

**Comment by luto** [comment]
@d-sko I can't find your mail address. Please write us an email to hallo@uberspace.de for your reward :)

**Pull request merged**